src/Disks/S3/DiskS3.cpp

```diff
@@ -247,8 +242,17 @@ std::unique_ptr<WriteBufferFromFileBase> DiskS3::writeFile(const String & path,
     auto settings = current_settings.get();
     auto metadata = readOrCreateMetaForWriting(path, mode);
 
-    /// Path to store new S3 object.
-    auto s3_path = getRandomASCIIString();
+    /// Path to store the new S3 object.
+
+    /// Total length is 32 a-z characters for enough randomness.
+    /// First 3 characters are used as a prefix for
+    /// https://aws.amazon.com/premiumsupport/knowledge-center/s3-object-key-naming-pattern/
+
+    constexpr size_t key_name_total_size = 32;
+    constexpr size_t key_name_prefix_size = 3;
+
+    auto s3_path = fmt::format("{}/{}",
```

Comment:
So we will have "random" prefixes for all objects?

Reply:
Yes, we simply put `/` after first three random letters.